examples/tv-app/linux/include/keypad-input/KeypadInputManager.cpp

```diff
@@ -51,22 +52,31 @@ EmberAfKeypadInputStatus KeypadInputManager::proxyKeypadInputRequest(EmberAfKeyp
     return EMBER_ZCL_KEYPAD_INPUT_STATUS_SUCCESS;
 }
 
-static void sendResponse(const char * responseName, EmberAfKeypadInputStatus keypadInputStatus)
+static void sendResponse(chip::app::Command * command, const char * responseName, EmberAfKeypadInputStatus keypadInputStatus)
 {
     static_assert(std::is_same<std::underlying_type_t<EmberAfKeypadInputStatus>, uint8_t>::value, "Wrong enum size");
-    emberAfFillExternalBuffer((ZCL_CLUSTER_SPECIFIC_COMMAND | ZCL_FRAME_CONTROL_SERVER_TO_CLIENT), ZCL_KEYPAD_INPUT_CLUSTER_ID,
-                              ZCL_SEND_KEY_RESPONSE_COMMAND_ID, "u", keypadInputStatus);
+    CHIP_ERROR err                         = CHIP_NO_ERROR;
+    chip::app::CommandPathParams cmdParams = { emberAfCurrentEndpoint(), /* group id */ 0, ZCL_KEYPAD_INPUT_CLUSTER_ID,
+                                               ZCL_SEND_KEY_RESPONSE_COMMAND_ID, (chip::app::CommandPathFlags::kEndpointIdValid) };
+    chip::TLV::TLVWriter * writer          = nullptr;
 
-    EmberStatus status = emberAfSendResponse();
-    if (status != EMBER_SUCCESS)
+    VerifyOrExit(command != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    SuccessOrExit(err = command->PrepareCommand(cmdParams));
+    VerifyOrExit((writer = command->GetCommandDataElementTLVWriter()) != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+    SuccessOrExit(err = writer->Put(chip::TLV::ContextTag(0), keypadInputStatus));
+    SuccessOrExit(err = command->FinishCommand());
+
+exit:
+    if (err != CHIP_NO_ERROR)
     {
-        ChipLogError(Zcl, "Failed to send %s. Error:%s", responseName, chip::ErrorStr(EMBER_ZCL_STATUS_NOT_AUTHORIZED));
+        ChipLogError(Zcl, "Failed to encode response command.");
     }
 }
 
-bool emberAfKeypadInputClusterSendKeyCallback(EmberAfKeypadInputCecKeyCode keyCode)
+bool emberAfKeypadInputClusterSendKeyCallback(chip::app::Command * command, uint8_t keyCode)
```

Comment:
Why this change?  The type in the XML is correct, so I would expect this to be getting `EmberAfKeypadInputCecKeyCode` as an argument.  Is it not?  @vivien-apple any idea what's going on here?

Reply:
If I leave EmberAfKeypadInputCecKeyCode as type I get a compile error:
`Undefined symbols for architecture x86_64:
  "emberAfKeypadInputClusterSendKeyCallback(chip::app::Command*, unsigned char)", referenced from:
      chip::app::clusters::KeypadInput::DispatchServerCommand(chip::app::Command*, unsigned int, unsigned short, chip::TLV::TLVReader&) in tv-common.IMClusterCommandHandler.cpp.o`

Reply:
Yeah, I see that the generated header also has `uint8_t`.  I'm just a bit surprised by it.

Reply:
> Why this change? The type in the XML is correct, so I would expect this to be getting `EmberAfKeypadInputCecKeyCode` as an argument. Is it not? @vivien-apple any idea what's going on here?

The current code generates very basic types for all the method signatures via `asUnderlyingZclType`. It needs to be updated to expose custom enums/bitmaps.